tests/testthat/test-convert_mae_se_assay_to_dt.R

```diff
@@ -197,3 +197,98 @@ test_that("convert_mae_assay_to_dt works as expected", {
   expect_equal(setdiff(colnames(dt_w), col_w), setdiff(colnames(dt_l), col_l))
   expect_equal(max(dt_l[, .N, by = "normalization_type"]$N), NROW(dt_w))
 })
+
+
+test_that("convert_se_assay_to_custom_dt works fine", {
+  json_path <- system.file(package = "gDRcomponents", "settings.json")
+  s <- get_settings_from_json(json_path = json_path)
+  
+  se <- get_synthetic_data("finalMAE_small")[[1]]
+  dt1 <- convert_se_assay_to_custom_dt(se, assay_name = "Metrics")
+  checkmate::expect_data_table(dt1, min.rows = 2, min.cols = 2)
+  expect_true(all(s$METRIC_WISH_LIST %in% names(dt1)))
+  dt2 <-
+    convert_se_assay_to_custom_dt(se, assay_name = "Metrics", output_table = "Metrics_raw")
+  checkmate::expect_data_table(dt2, min.rows = 2, min.cols = 2)
+  expect_true(all(s$METRIC_WISH_LIST %in% names(dt2)))
+  dt3 <-
+    convert_se_assay_to_custom_dt(se, assay_name = "Metrics", output_table = "Metrics_initial")
+  checkmate::expect_data_table(dt3, min.rows = 2, min.cols = 2)
+  expect_false(identical(dt2, dt3))
+  checkmate::expect_data_table(dt2, min.rows = 2, min.cols = 2)
+  expect_true(all(c("x_mean", "x_AOC", "x_AOC_range", "xc50", "x_max", "ec50", 
+                    "x_inf", "x_0", "h", "r2", "x_sd_avg", "fit_type") %in% names(dt3)))
+  dt4 <- convert_se_assay_to_custom_dt(se, assay_name = "Averaged")
+  checkmate::expect_data_table(dt2, min.rows = 2, min.cols = 2)
+  expect_true(
+    all(c("GR value", "Relative Viability", "Std GR value", "Std Relative Viability") %in% names(dt4)))
+  dt5 <- convert_se_assay_to_custom_dt(se, assay_name = "Averaged", output_table = "Metrics")
+  expect_true(identical(dt4, dt5))
+  
+  se2 <-
+    gDRutils::get_synthetic_data("finalMAE_combo_matrix")[[1]]
```

Comment:
Redundant namespacing here and other places in this script `gDRutils::`

Reply:
✔️

Reply:
✔️